compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -287,6 +287,9 @@ pub fn eval_to_allocation_raw_provider<'tcx>(
             if let Some(error_reported) = tcx.typeck_opt_const_arg(def).tainted_by_errors {
```

Comment:
A bit more than planned for this PR, but if you want to try doing this cleanup in this PR, I'm not going to complain ;)

Basically remove the entire logic in the `is_local` if block and move whatever is missing in `load_mir` to it.

We can also guard the checks in `load_mir` with the is_local check, as any mir we load from dependencies must be error free

Reply:
That said, feel free not to do this in this PR .we can merge it soon in any case